### PR TITLE
Add python 3.11 testing

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Review by dcermak — CHANGES_REQUESTED** [review]
This needs adjustments in `tests/test_metadata.py` as well and a rebase.

**Review by dcermak — APPROVED** [review]

File: bci_tester/data.py [hunk 1]
```diff
@@ -258,8 +258,18 @@ def create_BCI(
 NODEJS_18_CONTAINER = create_BCI(build_tag="bci/nodejs:18")
 
 PYTHON36_CONTAINER = create_BCI(build_tag="bci/python:3.6")
```
Comment (jlausuch) [inline bci_tester/data.py:260]: shouldn't the `available_versions` also specified for 3.6?

Comment (dirkmueller) [reply bci_tester/data.py:260]: I think the 3.6 container is available for any SP, so omitting it should be fine.

**Comment by dcermak** [comment]
[8983f54](https://github.com/SUSE/BCI-tests/pull/223/commits/8983f5427e03981f1ae4de37d4f0609d6b5e54e0) will require a big change in the settings to adjust the changed names of the expected CI checks…

**Comment by dirkmueller** [comment]
> [8983f54](https://github.com/SUSE/BCI-tests/pull/223/commits/8983f5427e03981f1ae4de37d4f0609d6b5e54e0) will require a big change in the settings to adjust the changed names of the expected CI checks…

I'm happy to do that once we get this reviewed and merged

**Comment by dcermak** [comment]
Dirk Mueller ***@***.***> writes:

>> [8983f54](https://github.com/SUSE/BCI-tests/pull/223/commits/8983f5427e03981f1ae4de37d4f0609d6b5e54e0) will require a big change in the settings to adjust the changed names of the expected CI checks…
>
> I'm happy to do that once we get this reviewed and merged

This has to be done *before* merging

File: bci_tester/data.py [hunk 2]
```diff
@@ -340,14 +350,26 @@ def create_BCI(
...
+CONTAINER_389DS_2_0 = create_BCI(
     build_tag="suse/389-ds:2.0",
+    available_versions=["15.4"],
+    bci_type=ImageType.APPLICATION,
+    healthcheck_timeout=timedelta(seconds=240),
+    extra_environment_variables={"SUFFIX_NAME": "dc=example,dc=com"},
+    forwarded_ports=[PortForwarding(container_port=3389)],
+)
+
+CONTAINER_389DS_2_2 = create_BCI(
```
Comment (dcermak) [inline bci_tester/data.py:362]: you need to add a marker to pyproject.toml for this

Comment (dirkmueller) [reply bci_tester/data.py:362]: done

**Review by dcermak — APPROVED** [review]

**Pull request merged**